packages/platform-android/native_modules.gradle

```diff
@@ -0,0 +1,249 @@
+import groovy.json.JsonSlurper
+import org.gradle.initialization.DefaultSettings
+
+def generatedFileName = "PackageList.java"
+def generatedFileContentsTemplate = """
+package com.facebook.react;
+
+import android.app.Application;
+import android.content.Context;
+import android.content.res.Resources;
+
+import com.facebook.react.ReactPackage;
+import com.facebook.react.shell.MainReactPackage;
+import java.util.Arrays;
+import java.util.List;
+
+{{ packageImports }}
+
+public class PackageList {
+  private ReactNativeHost reactNativeHost;
+  public PackageList(ReactNativeHost reactNativeHost) {
+    this.reactNativeHost = reactNativeHost;
+  }
+
+  private ReactNativeHost getReactNativeHost() {
+    return this.reactNativeHost;
+  }
+
+  private Resources getResources() {
+    return this.getApplication().getResources();
+  }
+
+  private Application getApplication() {
+    return this.reactNativeHost.getApplication();
+  }
+
+  private Context getApplicationContext() {
+    return this.getApplication().getApplicationContext();
+  }
+
+  public List<ReactPackage> getPackages() {
+    return Arrays.<ReactPackage>asList(
+      new MainReactPackage(){{ packageClassInstances }}
+    );
+  }
+}
+"""
+
+class ReactNativeModules {
+  private Logger logger
+  private Project project
+  private DefaultSettings defaultSettings
+  private ExtraPropertiesExtension extension
+  private ArrayList<HashMap<String, String>> reactNativeModules
+
+  private static String LOG_PREFIX = ":ReactNative:"
+  private static String REACT_NATIVE_CONFIG_CMD = "react-native config"
```

Comment:
This will execute config through global `react-native-cli`. Since we're aiming to get rid of the global package (or at least make users forget about it), please change it to a "require" from Node, like in CocoaPods implementation:
https://github.com/react-native-community/cli/blob/7d9f8cd96f8e5abc5b7c517d02fb166f9e51b877/packages/platform-ios/native_modules.rb#L3-L4

Reply:
Sure, give me a few minutes.

Reply:
Done 👍